Documentation/troubleshooting.md

```diff
@@ -63,6 +63,10 @@ Note: The `ServiceMonitor` references a `Service` (not a `Deployment`, or a `Pod
 kubectl -n monitoring get secret prometheus-k8s -ojson | jq -r '.data["prometheus.yaml.gz"]' | base64 -d | gunzip | grep "my-service-monitor"
 ```
 
+#### It is in the configuration but not on the Service Discovery page
+
+ServiceMonitors pointing to a Services that do not exist (e.g. nothing matching `.spec.selector`) will lead to this ServiceMonitor not being added to the Service Discovery page. Check if you can find any Service with the selector you configured.
+
```

Comment:
suggestion: we can add the following command which would work for service monitors that don't work cross-namespace and use "simple" matchLabels.

```bash
SERVICE_MONITOR_NAME="<to be replaced>" kubectl get services -l "$(kubectl get servicemonitors $SERVICE_MONITOR_NAME -o jsonpath="{.spec.selector.matchLabels}" | jq -cr 'to_entries | map(.key + "=" + .value) | join(",")')"
````

Reply:
added with some describing text, but without depending on any particular shell or `jq` being present

Reply:
Nice!